INV4/pallet-ips/src/tests.rs

```diff
@@ -200,12 +208,414 @@ fn destroy_should_fail() {
+
+        assert_eq!(Ips::next_ips_id(), 0);
+        assert_ok!(Ips::create_ips(
+            Origin::signed(ALICE),
+            MOCK_METADATA.to_vec(),
+            vec![0],
+            false,
+        ));
+
+        assert_ok!(Ips::allow_replica(
+            Origin::signed(multi_account_id::<Runtime, IpsId>(0, None)),
+            0
+        ));
+
+        assert_eq!(
+            IpsStorage::<Runtime>::get(0),
+            Some(IpsInfoOf::<Runtime> {
+                allow_replica: true,
+                parentage: Parentage::Parent(multi_account_id::<Runtime, IpsId>(0, None)),
+                metadata: MOCK_METADATA.to_vec().try_into().unwrap(),
+                data: vec![AnyId::IpfId(0)].try_into().unwrap(),
+                ips_type: IpsType::Normal
+            })
+        )
+    })
+}
+
+#[test]
+fn allow_replica_should_fail() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(Ipf::mint(
+            Origin::signed(ALICE),
+            MOCK_METADATA.to_vec(),
+            H256::from(MOCK_DATA)
+        ));
+
+        assert_eq!(Ips::next_ips_id(), 0);
+        assert_ok!(Ips::create_ips(
+            Origin::signed(ALICE),
+            MOCK_METADATA.to_vec(),
+            vec![0],
+            false,
+        ));
+
+        // Case 0: Extrinsic called in a non-multisig context:
+        assert_noop!(
+            Ips::allow_replica(Origin::signed(ALICE), 0),
+            Error::<Runtime>::NoPermission
+        );
+
+        // Case 1: An unknown origin tries to allow replica on IPS 0:
+        assert_noop!(
+            Ips::allow_replica(Origin::none(), 0),
+            DispatchError::BadOrigin,
+        );
+
+        assert_eq!(IpsStorage::<Runtime>::get(0).unwrap().allow_replica, false);
+    })
+}
+
+#[test]
+fn append_should_work() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(Ipf::mint(
+            Origin::signed(ALICE),
+            MOCK_METADATA.to_vec(),
+            H256::from(MOCK_DATA)
+        ));
+
+        assert_ok!(Ipf::mint(
+            Origin::signed(ALICE),
+            MOCK_METADATA_SECONDARY.to_vec(),
+            H256::from(MOCK_DATA_SECONDARY),
+        ));
+
+        assert_eq!(Ips::next_ips_id(), 0);
+        assert_ok!(Ips::create_ips(
+            Origin::signed(ALICE),
+            MOCK_METADATA.to_vec(),
+            vec![0],
+            true,
+        ));
+
+        assert_eq!(Ips::next_ips_id(), 1);
+        assert_ok!(Ips::create_replica(Origin::signed(ALICE), 0));
+
+        assert_ok!(Ips::append(
+            Origin::signed(multi_account_id::<Runtime, IpsId>(0, Some(ALICE))),
+            0,
+            vec![AnyId::IpfId(1)],
+            None
+        ));
+
+        assert_ok!(Ips::append(
+            Origin::signed(multi_account_id::<Runtime, IpsId>(
+                0,
+                Some(multi_account_id::<Runtime, IpsId>(1, None))
+            )),
+            0,
+            vec![AnyId::IpsId(1)],
+            None
+        ));
+
```

Comment:
You are missing the IPT checks, when you append an IPS it should burn the IPTs of the child and mint an exact copy in the parent

Reply:
Resolved on 35b037c + f3b81de